### PR TITLE
Move model bundles into Documents/models

### PR DESCRIPTION
**Pull request by nkashy1**
The new behaviour is that the application checks when it is loaded to
see if a Documents/models directory already exists. If not, it copies
over the initial models/ directory in which the build-time bundles are
present.

This sets up for future work in which we will let users download model
bundles directly onto their devices (without requiring them to be
present at build time).

- - -

Intended to resolve https://github.com/doc-ai/net-runner-ios/issues/6

@phildow : Please take a look. I couldn't find any unit tests in the `Net RunnerTests` directory to copypasta into a unit test for this behaviour, but I tested it in simulator (on iPhone X) and it worked.

### CODE REVIEW

**Comment by nkashy1** [comment]
I also tested the negative case of passing a bad model path to the `TIOBundleManager`.

The application correctly fails as expected.

File: Net Runner/AppDelegate.mm [hunk 1]
```diff
@@ -34,11 +34,25 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     
     assert( sizeof(float_t) == 4 );
     
-    // Load model bundles
-    
-    NSString *modelsPath = [[NSBundle mainBundle] pathForResource:@"models" ofType:nil];
+    // Move model bundles into Documents/ directory and then load them
```
Comment (phildow) [inline Net Runner/AppDelegate.mm]: This is all the right way to do it. `URLsForDirectory` and grabbing the zeroth index is correct, as odd as it looks. And going back and forth between URLs and String paths is sometimes necessary if annoying.

You could change:

1. moving the calls into separate methods on the AppDelegate class to clean it up
2. then you can also drop variables that are only used once and embed those calls in the other calls
3. use `URLByAppendingPathComponent` instead of getting the `path` and then calling `stringByAppendingPathComponent` on that.

It's stupid that `NSFileManager` doesn't have a `fileExistsAtURL` method, but you could also use the `fileExistsAtPath:` version without checking if it's a directory, since you don't actually check the value of `checkDirectory` anyway :)

Something like:

```objc
- (NSString*)appModelsPath {
  return [[NSBundle mainBundle] pathForResource:@"models" ofType:nil];
}

- (NSString*)documentsModelsPath {
  NSURL *documentDirectoryURL = [[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains: NSUserDomainMask][0];
  return [[documentDirectoryURL URLByAppendingPathComponent:@"models"] path];
}

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  // ...

  NSFileManager *fileManager = [NSFileManager defaultManager];

  if ( ![fileManager fileExistsAtPath:[self documentsModelsPath] ) {
    NSError *error;
    if ( ![fileManager copyItemAtPath:[self appModelsPath] toPath:[self documentsModelsPath] error:&error] ) {
      NSLog(@"Error copying models: %@", error);
    }
  }
}
```

It ends up making some of the calls more than once, but the user is never going to notice the nanoseconds worth of difference, and it's more readable.
